### PR TITLE
update ats tests

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Towards https://github.com/giantswarm/roadmap/issues/3612

This PR updates the tests run by ats so that those tests are run before pushing the app to all collections. The tests only check the following : 

- api-server availability
- grafana deployment present
- grafana pod running and ready

### CODE REVIEW

**Comment by QuantumEnigmaa** [comment]
All tests passing

File: tests/ats/test_grafana.py [hunk 1]
```diff
@@ -1,16 +1,69 @@
...
+def test_api_working(kube_cluster: Cluster) -> None:
+    """Very minimalistic example of using the [kube_cluster](pytest_helm_charts.fixtures.kube_cluster)
+    fixture to get an instance of [Cluster](pytest_helm_charts.clusters.Cluster) under test
+    and access its [kube_client](pytest_helm_charts.clusters.Cluster.kube_client) property
+    to get access to Kubernetes API of cluster under test.
+    Please refer to [pykube](https://pykube.readthedocs.io/en/latest/api/pykube.html) to get docs
+    for [HTTPClient](https://pykube.readthedocs.io/en/latest/api/pykube.html#pykube.http.HTTPClient).
+    """
+    assert kube_cluster.kube_client is not None
+    assert len(pykube.Node.objects(kube_cluster.kube_client)) >= 1
```
Comment (hervenicol) [inline tests/ats/test_grafana.py:26]: Do we need this test?
According to the comments it's just supposed to be an example.

File: tests/test-values.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      searchNamespace: ALL
+    datasources:
+      enabled: true
+      label: app.giantswarm.io/kind
+      labelValue: datasource
+      resource: both
+      searchNamespace: ALL
+    image:
+      repository: giantswarm/k8s-sidecar
+
```
Comment (hervenicol) [inline tests/test-values.yaml:113]: useless line

**Comment by hervenicol** [comment]
That looks really great! :astonished: :+1:

File: tests/ats/test_grafana.py (hunk 1 above)
Comment (QuantumEnigmaa) [reply tests/ats/test_grafana.py:26]: Not absolutely necessary but still nice to make sure that the apiserver is reachable before proceeding with any further tests right ?

File: tests/test-values.yaml (hunk 2 above)
Comment (QuantumEnigmaa) [reply tests/test-values.yaml:113]: Yeah, I just retrieved `grizzly`'s values and removed troublesome parts. 
Do you refer to only lines from 110 to 112 or to the whole sidecar section ?

Comment (hervenicol) [reply tests/test-values.yaml:113]: I'm only complaining about the empty line at the end of the file.

File: tests/ats/test_grafana.py (hunk 1 above)
Comment (hervenicol) [reply tests/ats/test_grafana.py:26]: Ok, why not, but then that's what the comment should say.

File: tests/test-values.yaml (hunk 2 above)
Comment (QuantumEnigmaa) [reply tests/test-values.yaml]: Oh ok. Removed it :)

File: tests/ats/test_grafana.py (hunk 1 above)
Comment (QuantumEnigmaa) [reply tests/ats/test_grafana.py]: Updated the comment.

**Comment by QuantumEnigmaa** [comment]
Maybe I should move the tests before the jobs pushing the app to the catalogs ? :thinking:

File: .circleci/config.yml [hunk 3]
```diff
@@ -31,6 +48,18 @@ workflows:
...
+          requires:
+            - "template-chart"
+      
+      - architect/run-tests-with-ats:
+          name: run-chart-tests-with-ats
+          filters:
+            # Do not trigger the job on merge to master.
+            branches:
+              ignore:
+                - master
```
Comment (QuentinBisson) [inline .circleci/config.yml:60]: You mean main right?

Comment (QuantumEnigmaa) [reply .circleci/config.yml:60]: I would have preferred but unfortunately, this repo's "main" branch is called "master" :/

Comment (QuentinBisson) [reply .circleci/config.yml:60]: Is it? Maybe we should change it then :D

Comment (QuantumEnigmaa) [reply .circleci/config.yml:60]: I would totally agree :)

File: tests/ats/test_grafana.py [hunk 4]
```diff
@@ -1,16 +1,65 @@
...
+def pods(kube_cluster: Cluster) -> List[pykube.Pod]:
+    pods = pykube.Pod.objects(kube_cluster.kube_client)
+
+    pods = pods.filter(namespace=namespace_name, selector={
+                       'app.kubernetes.io/name': 'grafana', 'app.kubernetes.io/instance': 'grafana'})
+
+    return pods
+
+# when we start the tests on circleci, we have to wait for pods to be available, hence
+# this additional delay and retries
```
[the remark this answers is not in the document]
Comment (QuantumEnigmaa) [reply tests/ats/test_grafana.py:56]: You're right. Just removed those lines.

**Review by QuentinBisson — APPROVED** [review]

**Pull request merged**